File: myimages_ui/src/services/LicenseService.js
import http from "../http-common"; 

class LicenseService {
  getAllLicenses(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/license/licenses`, searchDTO);
  }
 

  get(licenseId) {
    return this.getRequest(`/license/${licenseId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/license?field=${matchData}`, null);
  }

  addLicense(data) {
    return http.post("/license/addLicense", data);
  }

  update(data) {
  	return http.post("/license/updateLicense", data);
  }
  
  uploadImage(data,licenseId) {
  	return http.postForm("/license/uploadImage/"+licenseId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new LicenseService();

File: src/main/java/com/myimages/service/LicenseService.java
package com.myimages.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myimages.domain.License;
import com.myimages.dto.LicenseDTO;
import com.myimages.dto.LicenseSearchDTO;
import com.myimages.dto.LicensePageDTO;
import com.myimages.dto.LicenseConvertCriteriaDTO;
import com.myimages.service.GenericService;
import com.myimages.dto.common.RequestDTO;
import com.myimages.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface LicenseService extends GenericService<License, Integer> {

	List<License> findAll();

	ResultDTO addLicense(LicenseDTO licenseDTO, RequestDTO requestDTO);

	ResultDTO updateLicense(LicenseDTO licenseDTO, RequestDTO requestDTO);

    Page<License> getAllLicenses(Pageable pageable);

    Page<License> getAllLicenses(Specification<License> spec, Pageable pageable);

	ResponseEntity<LicensePageDTO> getLicenses(LicenseSearchDTO licenseSearchDTO);
	
	List<LicenseDTO> convertLicensesToLicenseDTOs(List<License> licenses, LicenseConvertCriteriaDTO convertCriteria);

	LicenseDTO getLicenseDTOById(Integer licenseId);


}
